### PR TITLE
Enables gRPC Health Checking Protocol on Open Telemetry Collector

### PR DESCRIPTION
**Pull request by pantuza**
**Description:** Adding a feature - Enabling [gRPC Health Checking Protocol](https://github.com/grpc/grpc/blob/master/doc/health-checking.md) on Open Telemetry Collector. 

**Link to tracking Issue:** Not Applicable

**Testing:**
Before this PR, if you use any gRPC tool [1,2,3] to probe otelcol gRPC port (4317) you would get as response:
```bash
$> grpc-health-probe -addr=localhost:4317
error: this server does not implement the grpc health protocol (grpc.health.v1.Health): unknown service grpc.health.v1.Health
```

Now, the gRPC server responds properly to gRPC Health Protocol:
```bash
$> grpc-health-probe -addr=localhost:4317
status: SERVING
```

* [1] [grpc-health-probe](https://github.com/grpc-ecosystem/grpc-health-probe)
* [2] [grpcurl](https://github.com/fullstorydev/grpcurl)
* [3] [ghz](https://github.com/bojand/ghz)

**Documentation:**
I will be glad to add documentation somewhere in the project. 
By now, I am considering these changes simply complies with the gRPC industry standard [gRPC Health Checking Protocol](https://github.com/grpc/grpc/blob/master/doc/health-checking.md). 
Thus, I think it worths an entry on the Changelog but not necessarily a documentation entry.

**Why these changes matter?**
It is quite common for gRPC users to probe their gRPC endpoints using the tools described above. 
It helps checking if your remote server is up and running before even knowing details of its remote routes/methods.

Therefore, this PR improves developer experience while using or getting started with Open Telemetry Collector. 
Here follow some of arguments for this PR:
1. It makes Collector more observable which complies with [this section](https://github.com/pantuza/opentelemetry-collector/blob/main/docs/vision.md#observable) of the long-term vision
2. It makes Collector more usable out of the box as described in [this section](https://github.com/pantuza/opentelemetry-collector/blob/main/docs/vision.md#usable-out-of-the-box) of the long-term vision
3. It adds a quite simple feature which complies with the [contributing guideline](https://github.com/pantuza/opentelemetry-collector/blob/main/CONTRIBUTING.md#report-a-bug-or-requesting-feature)
4. It does not add ANY new dependency. Basically we simply enabled a feature that was sitting in the project but not getting used. The code uses `google.golang.org/grpc` library which was already imported in the Collector code. 

Please let me know if there is anything missing here or anything I can improve in this contribution.

### CODE REVIEW

**Review by yurishkuro — APPROVED** [review]
+1

**Comment by atoulme** [comment]
I triggered them for you.

**Comment by pantuza** [comment]
Hi folks, I have added the changelog entry (file). 
Can we trigger CI once again to check if everything is ok?

File: .chloggen/grpc-health-checking-protocol.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. otlpreceiver)
+component: otlpreceiver
```
Comment (TylerHelmuth) [inline .chloggen/grpc-health-checking-protocol.yaml:7]:
```suggestion
component: configgrpc
```
This is technically a feature of anything that uses `configgrpc`.

File: .chloggen/grpc-health-checking-protocol.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: "Enable gRPC Health Checking Protocol on Open Telemetry Collector OTLP Receiver"
```
Comment (TylerHelmuth) [inline .chloggen/grpc-health-checking-protocol.yaml:10]:
```suggestion
note: "Enable gRPC Health Checking Protocol for any component that uses `configgrpc`, such as the Open Telemetry Collector OTLP Receiver"
```

**Review by TylerHelmuth — APPROVED** [review]

**Comment by pantuza** [comment]
Thank you for the suggestions, @TylerHelmuth. 
I have also updated this PR with latest changes that was on main branch.
Do you mind triggering the CI so we can check if there is anything missing? 
Please, let me know if you see any other improvements.

**Comment by TylerHelmuth** [comment]
@pantuza I don't have the right permissions in this repo. @open-telemetry/collector-approvers please take a look.

**Comment by atoulme** [comment]
Is this a dup of https://github.com/open-telemetry/opentelemetry-collector/pull/8397 ?

**Comment by pantuza** [comment]
> Hello @pantuza, Is there anything that might be changed from your side to pass checks?
> 
> FYI, @open-telemetry/collector-approvers, @dmitryax

Hi, @ksandrmatveyev. Thank you for the reply. 
Not actually. I don't have permissions to trigger the CI execution. Last time it was triggered, everything passed. 

I see that by now this PR is outdated from Master/Main branch. 
I will make sure to update it. 

I would be really glad If someone can trigger the CI so we can check if everything is ok.

**Comment by TylerHelmuth** [comment]
@pantuza I'm adding this PR to the collector's SIG agenda on Monday.

**Comment by jpkrohling** [comment]
> is there any reason a user wouldn't want to have this enabled?

In general, it's good practice to have admin-related endpoints at a different port than the workload, but for this specific one, I think it's ok and desirable to have it the way it's implemented in this PR.

**Comment by ksandrmatveyev** [comment]
Seems it fails on test phase https://github.com/open-telemetry/opentelemetry-collector/actions/runs/8838230921/job/24299483330?pr=8955  FYI @pantuza

**Comment by pantuza** [comment]
> Seems it fails on test phase https://github.com/open-telemetry/opentelemetry-collector/actions/runs/8838230921/job/24299483330?pr=8955 FYI @pantuza

Hi, @ksandrmatveyev . Thank you for the reply. 

Looking at the error, it is related to [contrib](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/82a828b798cd9074b9a3aec95c2115e825042799/extension/jaegerremotesampling/internal/grpc.go#L69-L71) project. That `jaegerremotesampling` extension is registering the Health server. Thus, gRPC fails saying there are two codes trying to register the same service. 

Since we are enabling it directly on the Collector, no extension needs to enable it. It will become a builtin feature of the 
Collector. What do you think? 

As a follow up for this PR, I can send a Pull Request to contrib project disabling this on `jaegerremotesampling` code. 
Any thoughts on my suggestions?

**Pull request closed without merging**